### PR TITLE
Added test for length mismatch panic in Correlation

### PR DESCRIPTION
**Pull request by Dawny33**

### CODE REVIEW

File: stat_test.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ func TestCorrelation(t *testing.T) {
 			t.Errorf("Correlation mismatch case %d. Expected %v, Found %v", i, test.ans, c)
 		}
 	}
+	if !Panics(func() { Correlation(make([]float64, 2), make([]float64, 3), make([]float64, 3)) }) {
```
Comment (btracey) [inline stat_test.go]: Given you're adding this, could you also add two more tests, one where `weights = nil` and one where `len(x) = len(y) != len(w)`. Thanks.

Comment (Dawny33) [inline stat_test.go]: @btracey Yeah, those should cover pretty nice.  Would add them too 👍

**Comment by Dawny33** [comment]
@btracey Added tests for `weights = nil` and weights length mismatch panics. Pl review.

**Comment by btracey** [comment]
LGTM

Could you submit a PR to add yourself to Contributors and Authors if necessary?

**Comment by Dawny33** [comment]
@btracey Done. Thanks :)  (https://github.com/gonum/license/pull/24)
